receiver/receiverhelper/scrapercontroller.go

```diff
@@ -58,63 +54,73 @@ func AddMetricsScraper(scraper MetricsScraper) ScraperControllerOption {
 // metrics will be passed to the next consumer.
 func AddResourceMetricsScraper(scrape ResourceMetricsScraper) ScraperControllerOption {
 	return func(o *scraperController) {
-		o.metricScrapers = append(o.metricScrapers, scrape)
+		o.resourceMetricScrapers = append(o.resourceMetricScrapers, scrape)
 	}
 }
 
 type scraperController struct {
-	defaultCollectionInterval time.Duration
-	nextConsumer              consumer.MetricsConsumer
+	collectionInterval time.Duration
+	nextConsumer       consumer.MetricsConsumer
 
-	metricScrapers []ResourceMetricsScraper
-	done           chan struct{}
+	metricsScrapers        *multiMetricScraper
+	resourceMetricScrapers []ResourceMetricsScraper
+	done                   chan struct{}
 }
 
 // NewScraperControllerReceiver creates a Receiver with the configured options, that can control multiple scrapers.
-func NewScraperControllerReceiver(_ configmodels.Receiver, nextConsumer consumer.MetricsConsumer, options ...ScraperControllerOption) (component.Receiver, error) {
+func NewScraperControllerReceiver(cfg ScraperControllerSettings, nextConsumer consumer.MetricsConsumer, options ...ScraperControllerOption) (component.Receiver, error) {
```

Comment:
You will need the Config for the Name of the receiver, to record metrics.

Reply:
Embedded `configmodels.ReceiverSettings` into `ScraperControllerSettings`

Also added an interface so that `ScraperControllerSettings` can be embedded into user config and updated comments to explain this.